photon-ml/src/main/scala/com/linkedin/photon/ml/util/PhotonLogger.scala

```diff
@@ -62,7 +72,28 @@ protected[ml] class PhotonLogger(logPath: Path, sc: SparkContext) extends Marker
    * clean up when done.
    */
   def close() {
-    writer.close()
+    try {
+      val fs = FileSystem.get(new URI(logPath.toString()), new Configuration())
+      writer.close()
+
+      val parent = logPath.getParent
+      // Create the parent directory if it doesn't exist already
+      if (!fs.exists(parent)) {
+        fs.mkdirs(parent)
+      }
+
+      // Overwrite any existing file
+      // TODO: make this configurable?
+      if (fs.exists(logPath)) {
+        fs.delete(logPath, false)
+      }
```

Comment:
As in #89, currently the logic in Photon is that if the output directory of Photon already exists, an error will be thrown. Since the ```logPath``` is always a child of Photon's output directory, in order to be consistent with the rest of code I guess we may not need the check and overwriting here.

Actually I forgot to remove this logic in PR#89 :)